src/main/resources/ItemDetails.yaml

```diff
@@ -0,0 +1,64 @@
+---
+bulkloadconfig:
+  samplefile : ItemDetails.xml
+  numberoffiles: 2
+  childnode: ItemPriceList
+  numberofmainnodes: 1
```

Comment:
Did you test the code with and without all these combinations?
  numberofmainnodes: 
  numberofchildnodes: 
  numberofsubchildnodes: